File: 507/perfect-number.js
/**
 * @param {number} num
 * @return {boolean}
 */
var checkPerfectNumber = function (num) {
    if (num == 1)
        return false;

    let sum = 1;
    let bound = ~~Math.sqrt(num);

    for (let i = 2; i <= bound; i++) {
        if (num % i == 0)
            sum += i + num / i;
    }

    return sum == num;
};

let num = 28;
console.log(checkPerfectNumber(num));

File: 507/perfect-number.go
package main

import (
	"fmt"
	"math"
)

func checkPerfectNumber(num int) bool {
	if num == 1 {
		return false
	}

	sum := 1
	bound := int(math.Sqrt(float64(num)))

	for i := 2; i <= bound; i++ {
		if num%i == 0 {
			sum += i + (num / i)
		}
	}

	return sum == num
}

func main() {
	num := 28
	fmt.Println(checkPerfectNumber(num))
}
